tests/CMakeLists.txt

```diff
@@ -86,6 +86,10 @@ elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     #json_test_set_test_options(test-disabled_exceptions COMPILE_DEFINITIONS _HAS_EXCEPTIONS=0 COMPILE_OPTIONS /EH)
 endif()
 
+# Suppress GCC<11 warning about unknown pragmas
+json_test_set_test_options(test-items COMPILE_OPTIONS
```

Comment:
Just to get it right: `-Wno-pragma` cannot be set by a pragma inside the sources?

Reply:
It could be done in the source but this option seemed less clumsy. Let me try to do it in the source and see what it takes to compare.

Reply:
We don't use `HEDLEY_*` because those aren't exported. But because of the doctest 2.4.9 issue I finally understand how doctest's compiler version macros work. We really should replace `#if defined(__GNUC__)` etc. with those.

It's a lot cleaner. I'll rename this PR and do it now all of this if that's OK with you?

Reply:
Yes, that would be great. Sorry for the confusion - I know we're not using Hedley - with Hedley-like I meant exactly that kind of abstraction.